### PR TITLE
update prerender to include a content manifest csv output

### PR DESCRIPTION
**Pull request by TomWoodward**
https://openstax.atlassian.net/browse/DISCO-23

adds a csv output of all book toc node info for use by reporting platforms to associate uuids with slugs with titles (for querying and output formatting in reporting dashboards)

also moved sitemap index output to queue generation to avoid loading all the books twice

### CODE REVIEW

File: script/prerender/sitemap.ts [hunk 1]
```diff
@@ -1,14 +1,11 @@
...
-import map from 'lodash/fp/map';
-import max from 'lodash/fp/max';
 import sitemap, { SitemapItemOptions } from 'sitemap';
 import { SerializedPageMatch } from './contentRoutes';
 import { writeAssetFile } from './fileUtils';
+import { BookWithOSWebData } from '../../src/app/content/types';
+import { getSitemapItemOptions } from './contentPages';
 
 export const sitemapPath = (pathName: string) => `/rex/sitemaps/${pathName}.xml`;
+export const contentManifestPath = (pathName: string) => `/rex/manifest/${pathName}.xml`;
```
Comment (jivey) [inline script/prerender/sitemap.ts:8]: I don't think this gets used?

**Review by jivey — APPROVED** [review]
Looks good just some minor things/questions

File: script/prerender/contentManifest.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      ? ['', '']
+      : [node.slug, content.getUrl({book: {slug: book.slug}, page: {slug: node.slug}})];
+
+  const contents = 'contents' in node
+    ? node.contents.map(child => getContentsRows(book, child, titleNumber || chapterNumber))
+      .reduce((result, item) => ([...result, ...item]), [] as string[][])
+    : [];
+
+  return [
+    [stripIdVersion(id), title, textTitle, book.language, ...urlParams, tocType, toc_target_type ?? ''],
```
[the remark this answers is not in the document]
Comment (TomWoodward) [reply script/prerender/contentManifest.ts:48]: Yea I doubt that the html will be used by anything but I threw it in there. the text title is intended to be used by reporting I added the context number in there for the eoc pages

File: script/prerender/fleet.ts [hunk 3]
```diff
@@ -43,11 +43,13 @@ import { getBooksConfigSync } from '../../src/gateways/createBookConfigLoader';
...
 import { globalMinuteCounter, prepareBookPages } from './contentPages';
-import { SerializedBookMatch, SerializedPageMatch } from './contentRoutes';
+import { SerializedPageMatch } from './contentRoutes';
 import createRedirects from './createRedirects';
 import './logUnhandledRejectionsAndExit';
 import renderManifest from './renderManifest';
-import { SitemapPayload } from './sitemap';
+import { SitemapPayload, renderAndSaveSitemapIndex } from './sitemap';
+import { writeS3ReleaseXmlFile } from './fileUtils';
+import { renderAndSaveContentManfiest } from './contentManifest';
```
Comment (jivey) [inline script/prerender/fleet.ts:52]: Import didn't get renamed

**Review by jivey — CHANGES_REQUESTED** [review]

**Review by jivey — APPROVED** [review]

**Pull request merged**